src/app/src/test/kotlin/org/imperial/mrc/hint/unit/controllers/ADRControllerTests.kt

```diff
@@ -305,6 +304,103 @@ class ADRControllerTests : HintrControllerTests()
         }
     }
 
+    @Test
+    fun `pushes output zip to ADR`()
+    {
+        val mockAPIClient: HintrAPIClient = mock {
+            on { downloadSpectrum("model1") } doReturn ResponseEntity.ok().body(StreamingResponseBody { it.write("".toByteArray()) })
+        }
+        val mockClient: ADRClient = mock {
+            on { postFile(eq("resource_create"), eq(listOf("name" to "output1.zip", "description" to "Naomi model outputs", "hash" to "D41D8CD98F00B204E9800998ECF8427E", "package_id" to "dataset1")), any()) } doReturn ResponseEntity.ok().body("whatever")
+        }
+        val mockBuilder: ADRClientBuilder = mock {
+            on { build() } doReturn mockClient
+        }
+        val sut = ADRController(mock(), mock(), mockBuilder, mock(), mockProperties, mock(), mockAPIClient, mock(), mock())
+        val result = sut.pushFileToADR("dataset1", "adr-output-zip", "model1", "output1.zip", null)
```

Comment:
I guess it would be good to test that the file parameter to `ADRClient.postFile` looks as expected, since that's just matching on `any()`. That's always super-fiddly though. ArgumentCaptor can be helpful.

Reply:
Good call - I've made a change which I assume is consistent with your thinking